### ISSUE TITLE
Improve `starknet-foundry` version for a future complete migration

### ISSUE THREAD

**Issue by jrcarlos2000** [Scaffold-Stark/scaffold-stark-2#241]
## Issue Overview

Currently [this branch](https://github.com/Quantum3-Labs/scaffold-stark-2/tree/starknet-foundry) supports deployment scripts fully written in cairo. It uses [Starknet foundry](https://github.com/foundry-rs/starknet-foundry)

Some pending TODOs are ( in order of priority ):

- In [deploy script](https://github.com/Quantum3-Labs/scaffold-stark-2/blob/starknet-foundry/packages/snfoundry/scripts/src/scripts.cairo), skip contract declaration if it exists, this is a pending fix from snfoundry.
- there are 2 `scarb.toml` files, since this all use the same versions of cairo and starknet, are we able to have only one `scarb.toml`, or one single project that has scripts, tests and contracts. This will def have a better UX.
- minimize the usage of [this script](https://github.com/Quantum3-Labs/scaffold-stark-2/blob/starknet-foundry/packages/snfoundry/scripts-ts/helpers/command.ts), its better if we can get rid of it.
   - Idea : Consider exploring the usage of environment variables like in [foundry](https://book.getfoundry.sh/reference/forge/forge-script?highlight=forge%20scr#forge-script), this will let us have an `.env` and avoid passing params like `--account` when running it.
- Currently transactions take long to execute, are we able to either:
   - choose the block tag to be pending by default
   - build the contract deployment transactions , batch them and execute at the end of the script, similar to [the current deploy script on `main`](https://github.com/Quantum3-Labs/scaffold-stark-2/blob/35efe94fb6e52434b5cc115eb7e43d7d5897445a/packages/snfoundry/scripts-ts/deploy-contract.ts#L48)
   - [IMPORTANT] the waiting time is considerably slower compared to scripts in `main`. In order to have this by default, the speed has to be same or faster.

**Comment by oceanlvr** [comment]
for issue 1:
> In [deploy script](https://github.com/Quantum3-Labs/scaffold-stark-2/blob/starknet-foundry/packages/snfoundry/scripts/src/scripts.cairo), skip contract declaration if it exists, this is a pending fix from snfoundry.

it's seems that `[Starknet foundry](https://github.com/foundry-rs/starknet-foundry)` we use is version `0.27.0`

and the latest version is `0.30.0`, we could use api like this to skip declare if it have been declared. it's seems that we could not now what if the contract have been declared in  `0.27.0`.could i upgrade this package? so that i could use this api

<img width="923" alt="image" src="https://github.com/user-attachments/assets/00ebd9d0-6169-4735-9486-e2b4369dde0c">
<img width="537" alt="image" src="https://github.com/user-attachments/assets/c82d1c92-3cf8-46fd-876e-3674a1455b21">


<img width="814" alt="image" src="https://github.com/user-attachments/assets/463cc3fa-e836-4e4c-a6ad-45cdfe233ad0">


https://foundry-rs.github.io/starknet-foundry/appendix/snforge-library/declare.html#declare

**Comment by metalboyrick** [comment]
reopen when needed
